services/static-webserver/client/source/class/osparc/study/CreateFunction.js

```diff
@@ -98,44 +98,7 @@ qx.Class.define("osparc.study.CreateFunction", {
       column = 0;
       row++;
 
-      const filePickers = osparc.study.Utils.extractFilePickers(this.__studyData["workbench"]);
-      filePickers.forEach(filePicker => {
-        const fpLabel = new qx.ui.basic.Label(filePicker["label"]);
-        inputsLayout.add(fpLabel, {
-          row,
-          column,
-        });
-        column++;
-
-        const fpType = new qx.ui.basic.Label("FileID");
-        inputsLayout.add(fpType, {
-          row,
-          column,
-        });
-        column++;
-
-        const fpExposed = new qx.ui.form.CheckBox().set({ value: true });
-        inputsLayout.add(fpExposed, {
-          row,
-          column,
-        });
-        exposedInputs[filePicker["label"]] = true;
-        fpExposed.addListener("changeValue", e => exposedInputs[filePicker["label"]] = e.getData());
-        column++;
-
-        const outputValue = osparc.file.FilePicker.getOutput(filePicker);
-        const fpDefaultValue = new qx.ui.basic.Label(outputValue && outputValue["path"] ? outputValue["path"] : "");
-        inputsLayout.add(fpDefaultValue, {
-          row,
-          column,
-        });
-        column++;
-
-        column = 0;
-        row++;
-      });
-
-      const parameters = osparc.study.Utils.extractParameters(this.__studyData["workbench"]);
+      const parameters = osparc.study.Utils.extractFunctionableParameters(this.__studyData["workbench"]);
```

Comment:
Maybe there is a particular reason to reimplement this, but wouldn't it make more sense to use an existing endpoint to get all the parameters of a study?

project_inputs = await webserver_api.get_project_inputs(project_id=project.uuid)

Reply:
- I wasn't even aware that that endpoint existed (``/inputs`` ``/outputs``)
- I would need those to be extended to return the also the type
- For this MVP, I wouldn't overload the webserver with two more calls per project to solve something that can be solved by a static functions in the frontend.

Reply:
👍